src/docs/binding-tuples.md

```diff
@@ -0,0 +1,60 @@
+# Binding Tuples
+
+Tuples in Swift are conceptually anonymous structs. Apple implies that in the documentation by saying that tuples follow the same packing rules as structs when they are stored in memory. However, this is absolutely not the case when tuples are being passed as arguments. For whatever reason, when passed as arguments tuples are flattened entirely to individual elements that are not tuples. In other words, the following two functions are identical from an ABI perspective:
+```swift
+public func foo(T: (a: Int, b: Bool, (c: SomeStruct, d: SomeClass))) { // func of 1 argument
+    // ...
+}
+
+public func foo(a: Int, b: Bool, c: SomeStruct, d: SomeClass) { // func of 4 arguments
+    // ...
+}
+```
+It's unclear why Apple would do this except that it might be a vestige of when Swift had the ability to have multiple
+argument lists and could do partial function application. It also might be because it enables very low-friction usage of tuples in that when used outside of the context of nominal types, tuples don't actually need to exist. They can be implemented as an artifical grouping of scoped variable names.
```

Comment:
Handling tuples in this manner allows Swift to enregister more parameters (as each element of the tuple can use up to 4 registers after physical lowering until registers are exhausted instead of the tuple as a whole only being able to use up to 4 registers).